public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx

```diff
@@ -11,20 +11,22 @@ import { LogMessages } from '../../Analytics';
 
 interface Props {
   className?: string;
-  queryString?: string;
   defaultQueryString?: string;
   onFilterChange: (filterString: string) => void;
 }
 
-export const MatcherFilter = ({ className, onFilterChange, defaultQueryString, queryString }: Props) => {
+export const MatcherFilter = ({ className, onFilterChange, defaultQueryString }: Props) => {
   const styles = useStyles2(getStyles);
+
   const handleSearchChange = debounce((e: FormEvent<HTMLInputElement>) => {
```

Comment:
Isn't `debounce` a culprit there? It creates a new `debounced` function on every render

Reply:
That's a fair point, it was a bit wonky but not the culprit – I'll still update this PR to use `useDebounce` instead